runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcGetDataStream.java

```diff
@@ -192,11 +205,35 @@ public void refreshActiveWork(Map<String, List<KeyedGetDataRequest>> active) {
                 .addRequests(request));
       }
     }
+    for (Map.Entry<String, List<HeartbeatRequest>> entry : heartbeats.entrySet()) {
+      for (HeartbeatRequest request : entry.getValue()) {
+        // Calculate the bytes with some overhead for proto encoding.
+        long bytes = (long) entry.getKey().length() + request.getSerializedSize() + 10;
+        if (builderBytes > 0
+            && (builderBytes + bytes > AbstractWindmillStream.RPC_STREAM_CHUNK_SIZE
+                || builder.getRequestIdCount() >= streamingRpcBatchLimit)) {
```

Comment:
we don't need the batch limit check here. we're not adding request ids for heartbeats

Reply:
Nice! I'll just remove all this then.